src/algorithms/reco/PrimaryVertices.cc

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Daniel Brandenburg
+
+#include <Math/GenVector/LorentzVector.h>
+#include <Math/GenVector/PxPyPzM4D.h>
+#include <Math/Vector4Dfwd.h>
+#include <edm4eic/VertexCollection.h>
+#include <edm4eic/unit_system.h>
+#include <edm4hep/Vector3f.h>
+#include <fmt/core.h>
+#include <podio/ObjectID.h>
+#include <iterator>
+#include <map>
+#include <utility>
+
+#include "algorithms/reco/PrimaryVertices.h"
+#include "algorithms/reco/PrimaryVerticesConfig.h"
+
+namespace eicrecon {
+
+  /**
+   * @brief Initialize the PrimaryVertices Algorithm
+   *
+   * @param logger
+   */
+  void PrimaryVertices::init(std::shared_ptr<spdlog::logger>& logger) {
+    m_log = logger;
+  }
+
+  /**
+   * @brief Produce a list of primary vertex candidates
+   *
+   * @param rcvtx  - input collection of all vertex candidates
+   * @return std::unique_ptr<edm4eic::VertexCollection>
+   */
+  std::unique_ptr<edm4eic::VertexCollection> PrimaryVertices::execute(
+                const edm4eic::VertexCollection *rcvtx
+        ){
+
+    // this map will store intermediate results
+    // so that we can sort them before filling output
+    // collection
+    std::map<int, edm4eic::Vertex> primaryVertexMap;
```

Comment:
```suggestion
    std::map<int, edm4eic::Vertex, std::greater<int>> primaryVertexMap;
```

Reply:
Updated. In the meantime, switch to use multimap to avoid vertices having the same N_trk number get dropped or replaced.